Add guides on diagrams and collapse blocks

tancnle opened the pull request:
# Description

Add some information on usages of Mermaid diagrams and Collapse blocks in Markdown.

Fixes https://github.com/thoth-tech/handbook/issues/65

## Type of change

- [x] Documentation (update or new)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

tancnle committed:
docs: add guides on diagrams and collapse blocks

talzid commented on docs/learning/training/markdown-guide.md:
```diff
@@ -63,28 +55,83 @@ Commonly used formatting options include the following:
+
+</details>
+
+To add markdown in the collapsed heading, be sure to add:
```
what do you mean by "add markdown in the collapsed heading"?  Maybe there is a different way to word it?

tancnle replied:
Sorry for the confusion @talzid. I was under the impression that markdown content can't be rendered inside the `<details>` block but I am wrong. Apparently, the following text,

```markdown
<details>

### Manificient header
</details>
```
can be rendered correctly as 

<details>

### Manificient header
</details>

I will remove this line 👍🏼

talzid reviewed (approved):
Looks good! Thank you for creating this Tan. 
I just left one comment cause I was confused by the wording.

tancnle committed:
docs: remove unnecessary markdown instruction

Markdow content can be rendered within collapsed heading without
`markdown` attribute.

tancnle committed:
docs: fix small typos in company name

kats8 reviewed (approved):
The Mermaid addition is cool, @tancnle Something new I have learned! :)
